trading_pair_azero/lib.rs

```diff
@@ -1800,18 +1810,27 @@ pub mod trading_pair_azero {
             value1: Balance,
             value2: Balance,
         ) -> Result<Balance, TradingPairErrors> {
-            let absolute_difference = value1.abs_diff(value2);
-
-            let absolute_difference_nominated = absolute_difference * (10u128.pow(12));
+            let mut percentage_difference: Balance = 0;
 
-            let percentage_difference: Balance;
+            if value1 > value2 {
+                percentage_difference = (value1 - value2) * (10u128.pow(12));
+                match (percentage_difference / value2).checked_mul(100u128) {
+                    Some(result) => {
+                        percentage_difference = result;
+                    }
+                    None => return Err(TradingPairErrors::Overflow),
+                };
+            }
 
-            match 100u128.checked_mul(absolute_difference_nominated / ((value1 + value2) / 2)) {
-                Some(result) => {
-                    percentage_difference = result;
-                }
-                None => return Err(TradingPairErrors::Overflow),
-            };
+            if value2 > value1 {
+                percentage_difference = (value2 - value1) * (10u128.pow(12));
+                match (percentage_difference / value1).checked_mul(100u128) {
```

Comment:
i think there is no need to multiply it by 100. It is already denominated with 12 zeroes. Not an error but not necessary.